bridge/src/main/java/com/livefront/bridge/BridgeDelegate.java

```diff
@@ -144,6 +145,26 @@ private boolean isAppInForeground() {
         return mActivityCount > 0 || mIsConfigChange;
     }
 
+    /**
+     * If it's a fresh start or back to main activity, we can safely clear cache
+     */
+    private boolean isFreshStartOrBackToMain(@NonNull Activity activity, Bundle savedInstanceState) {
+        if (savedInstanceState != null) {
+            return false;
+        }
+        ActivityManager systemService = (ActivityManager) activity.getSystemService(Context.ACTIVITY_SERVICE);
+        if (systemService == null) {
+            return false;
+        }
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
```

Comment:
👍

Reply:
Should this check be for Lollipop actually? The API was added in 21.

Reply:
numActivities is starting from 23 actually..

Reply:
Ah OK. I thought it all would have been there when the `getAppTasks()` method was added in 21 but apparently not.